File: backend/src/Assignment.Domain/Entities/Course.cs
namespace Assignment.Domain.Entities;

public class Course
{
    public Guid Id { get; set; }

    public string Code { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public Guid TeacherId { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/types/course.ts
export interface Course {
  id: string;
  code: string;
  name: string;
  description: string;
  teacherId: string;
  createdAt: string;
}

export interface CourseRequest {
  code: string;
  name: string;
  description: string;
  teacherId: string;
}
